### ISSUE TITLE
Incorrect spacing between 'Show more' and 'Show comments'

### ISSUE THREAD

**Issue by katbow** [dideler/toggle-youtube-comments#2]
On some videos there is a `margin-left:0` style applied to the 'Show comments' button, causing it to be very close to the 'Show more' button

Eg. from https://www.youtube.com/watch?v=QKzobTCIRDw
![image](https://user-images.githubusercontent.com/1287388/31780371-6ee30d88-b4ed-11e7-9cff-bd15108d1590.png)

```html
<button class="fake-paper-button" id="toggle-comments" style="margin-left:0" type="button">
  <span class="fake-yt-formatted-string">Show comments</span>
</button>
```

This doesn't apply to all videos, however. On [this video](https://www.youtube.com/watch?v=GVDJ8O3lPBA) and most others, the correct styling applies (`margin-left: 1em;`).
![image](https://user-images.githubusercontent.com/1287388/31780464-bc9d1c30-b4ed-11e7-854a-f8672c610430.png)

```html
<button class="fake-paper-button" id="toggle-comments" type="button">
  <span class="fake-yt-formatted-string">Show comments</span>
</button>
```

**Comment by katbow** [comment]
After looking in to this further, this is happening because:

After the first injection, the `eventListener` on the info container is removed so that multiple injections do not occur when navigating to new videos. This means that if you initially go to a video without a `Show more` button, the `margin-left` will be zero. Then if you click on another video that does have a `Show more` button, this margin will still apply since the `Show comments` button isn't re-added. This applies in reverse as well (i.e. show more button exists, a `margin-left` of `1em` will exist, and going to a video without a show more button will mean it's misaligned.

**Comment by katbow** [comment]
I'm going to look at if there is an event for when a new video is chosen, as this action doesn't render the page again. Then if there is, we can listen for this event and fire the inject function again.

**Comment by 1natsu172** [comment]
This issue still occurs… 😱 

The cause is the verification of @ katbow.
It seems that YouTube is a virtual DOM SinglePageApplication.
This is because Chrome extensions are not reloaded when page transitions occur.

Since the correction proposal(#3 ) has `!important`, gaps are created on SHOW MORE hidden pages.

<img width="394" alt="2017-12-29t01-27-32" src="https://user-images.githubusercontent.com/7282145/34418161-c8c333d0-ec3f-11e7-8f82-744b831c1072.png">

> … 🤔 

## Countermeasure

1. Make sure to reload Chrome extensions when transitioning pages.

For example, use like `chrome.tabs.onUpdated.addListener`.
But I think that this is heavy work because  it needs change to be a  background extension. 🔨 

2. Add margin to "SHOW MORE" button

Currently, "margin-left" is attached to "SHOW COMMENTS", but instead of "SHOW MORE" attach margin-right.
Even when the "SHOW MORE" button is not displayed, the DOM exists.
If this measure is the current specification, I think that Issue can be solved promptly.

***

My recommendation is 2's SHOW MORE margin-right method.

```javascript
_addButton() {
    debugLog('ADDING BUTTON...');
    const moreButton = document.getElementById('more');
    const moreButtonStyling = moreButton.style.marginRight = '1em'
    const button = `
    <button class="fake-paper-button" id="toggle-comments" type="button">
      <span class="fake-yt-formatted-string">${globals.showComments}</span>
    </button>
    `;
```
> Like this ☝️ 

What should we do? 💭

**Comment by 1natsu172** [comment]
I thought that ShadyDOM caused a fundamental problem, but it seems not. 🤔 

For this Issue, the hidden property is added dynamically depending on the page type.
And since the timing of property addition is unstable, injection of toggleButton is done before property addition.
Apparently this seems to be a fundamental problem.

<img width="729" alt="2017-12-30t01-38-35" src="https://user-images.githubusercontent.com/7282145/34442949-06c595a0-ed09-11e7-8629-d1d883e6067b.png">

***

Even if support to ShadyDOM, modification is necessary.
I think need to do either "SHOW MORE button's style" or "Wait for the addition of hidden property" ⚙️

**Comment by dideler** [comment]
Didn't get a notification for this activity, so I'm only seeing these comments now :confused:.

I like the simple approach of styling the SHOW MORE button instead of the SHOW COMMENTS button :+1:

**Comment by 1natsu172** [comment]
@dideler 

I recommended a style to the "SHOW MORE" button, but there was a defect… :man_facepalming: 

![2018-01-16 0 11 31](https://user-images.githubusercontent.com/7282145/34950484-a3ff919a-fa56-11e7-9da1-8019b7b84ea5.png)

But I found a best practice! 🤗 
Use adjacent sibling connectors!

```
#less[hidden] + #more[hidden] + .fake-paper-button {
    margin-left: 0;
}
```

![2018-01-16 0 13 19](https://user-images.githubusercontent.com/7282145/34950646-1b17a7c2-fa57-11e7-91d5-5d3c98f45839.png)

Extension should be able to provide stable buttons by applying the appropriate style to the "SHOW COMMENTS" button.

By the way, why am I connecting "[hidden]" by double, because there is a problem when user expand the description by click the "SHOW MORE" button.

![2018-01-16 0 04 50](https://user-images.githubusercontent.com/7282145/34951023-42eaaf00-fa58-11e7-8ad0-77c6315b2e59.png)

**Comment by dideler** [comment]
That's very cool, I didn't know adjacent sibling combinators existed!

**Comment by 1natsu172** [comment]
Fixed this issue by #29 🔫
